### ISSUE TITLE
How to add post processing after ‘make build’

### ISSUE THREAD

**Issue by weiweihuanghuang** [googlefonts/gftools#616]
What’s the best practise for adding post processing into the build process when working with the gftools font builder. For example if I want to autohint with psautothint after I let gftools build OTFs with fontmake? And is that something I can integrate with GitHub actions?

**Comment by m4rc1e** [comment]
I guess you could make a shell script which calls the builder first then does your other post processing steps.

Does the psautohinter improve the results on modern systems? I'm curious because I just want hinting to die.

**Comment by guidoferreyra** [comment]
I was thinking on this just today. I find the builder really great but I like to PS autohint otfs so I run a script after the builder to patch. I think it would be awesome to have PS Autohinting included.

I don’t have the best evidence, but see unhinted(bottom) vs autohinted (top).

![imagen](https://user-images.githubusercontent.com/1755590/195703220-d7da3f68-ce01-4b1c-8fe8-631b18334a68.png)

**Comment by guidoferreyra** [comment]
Made a fork and added psautohinting is working but not sure if its a good implementation. https://github.com/guidoferreyra/gftools/commit/2289ae6c566ea496134779bc6c52ffedd978fa7d

**Comment by m4rc1e** [comment]
@guidoferreyra which platform and software are the screenshots from?

**Comment by guidoferreyra** [comment]
That was on Illustrator on macOs. I didn’t tested on windows but I assume it will look worst

**Comment by m4rc1e** [comment]
Interesting. Do PS hints cause any difference in Mac apps? afaik ttf hints do nothing.

**Comment by guidoferreyra** [comment]
Yes, like you said, ttf hints do nothing but ps hinting is needed for supressing overshoots, making bold same pixel height than regulars, etc.

**Comment by m4rc1e** [comment]
Sorry let me rephrase. Do OS X apps such as TextEdit use PS font hints? we've established that Adobe apps do use PS hints. I'm now interested if Apple Apps use em.

**Comment by guidoferreyra** [comment]
oh, I got it now. No, Mac apps doesn’t use ps hints.

**Comment by weiweihuanghuang** [comment]
Is it worth making a pull request into gftools?

**Comment by guidoferreyra** [comment]
I don’t think my implementation is clean, it works but it doesn’t have the enough quality for making a PR

**Comment by m4rc1e** [comment]
A link to some code is fine I can clean it up. I'm still pondering whether we should autohint otfs. We've always left these unhinted

**Comment by guidoferreyra** [comment]
> Made a fork and added psautohinting, is working but not sure if it’s a good implementation. [guidoferreyra@2289ae6](https://github.com/guidoferreyra/gftools/commit/2289ae6c566ea496134779bc6c52ffedd978fa7d)

There is my commit. Maybe include it and make psautohint false by default?

**Comment by m4rc1e** [comment]
Thanks, I'll take a look tomorrow.

**Comment by m4rc1e** [comment]
@guidoferreyra it looks pretty good. I'll have a play with it and perhaps integrate. If we do include it, we'll make it false by default as you suggested.
